subsys/bluetooth/services/dis.c

```diff
@@ -32,6 +32,11 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(bt_dis);
 
+BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_DIS_MODEL_NUMBER) ||
+	     !IS_ENABLED(CONFIG_BT_DIS_MODEL_DEPRECATED_USED));
+BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_DIS_MANUF_NAME) ||
+	     !IS_ENABLED(CONFIG_BT_DIS_MANUF_DEPRECATED_USED));
```

Comment:
Maybe we could build assert only if `CONFIG_BT_DIS_MANUF_NAME_STR` does not match `CONFIG_BT_DIS_MANUF`? (that seems like the only case that is actually problematic for the implementation)

Reply:
Alternatively, maybe a Kconfig dependencies could prevent the user from using new options if deprecated ones are still in use